### PR TITLE
Codeception Gherkin Experiment

### PR DESCRIPTION
**Pull request by puneet0191**
Pull Request for Issue # .

### CODE REVIEW

**Comment by puneet0191** [comment]
@pritalpatel  Please review this, tell me what I need to improve :smile:

**Comment by pritalpatel** [comment]
Thanks @puneet0191 for PR. As you are using table and steps in gherkin is very useful in my further work. I will use this example and improve my PR.

File: tests/composer.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
 	"php": ">=5.3.10"
   },
   "require-dev": {
-	"codeception/codeception": "master@dev",
+	"codeception/codeception": "~2.2",
```
Comment (pritalpatel) [inline tests/composer.json:23]: I have sent separate PR for the same.

**Comment by puneet0191** [comment]
@pritalpatel feel free to use all the code, this is a sample PR, it will not be merged, just for our reference. I am also learning like you :dancer:

File: tests/_support/Step/Acceptance/Administrator/ContentSteps.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	public function iFillMandatoryFieldsForCreatingArticle(\Behat\Gherkin\Node\TableNode $fields)
+	{
+		$I = $this;
+		// iterate over all rows
+		foreach ($fields->getRows() as $index => $row) {
+			if ($index === 0) { // first row to define fields
+				$keys = $row;
+				continue;
+			}
+			else
```
Comment (yvesh) [inline tests/_support/Step/Acceptance/Administrator/ContentSteps.php]: This code could be improved ;)

...
}
elseif ($row[0] == 'title') 
{
...
}
elseif ($row[0] == 'content') 
{
...
}

Instead of the else.

$row[0] is also not good readable, it would be nice when you could do the following above:

$key = $row[0];
$value = $row[1];

That would improve readability

Comment (puneet0191) [inline tests/_support/Step/Acceptance/Administrator/ContentSteps.php]: Thanks, Makes sense

File: tests/_support/Step/Acceptance/Administrator/LoginSteps.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+class LoginSteps extends \AcceptanceTester
+{
+
+	private $usernameFieldId = ['id' => 'mod-login-username'];
+
+	private $passwordFieldId = ['id' => 'mod-login-password'];
+
+	private $pageTitleClass = ['class' => 'page-title'];
+
+	private $loginButtonPath = ['xpath' => "//button[contains(normalize-space(), 'Log in')]"];
```
Comment (yvesh) [inline tests/_support/Step/Acceptance/Administrator/LoginSteps.php]: That xpath is probably really slow :( As we don't have a unique identifier there, we probably should try using the name ($I->click('Log in');) What do you think?

Or you can use $I->click('class' => 'btn-primary'); here.. Even if it is generic, it's the only one on that page.

File: tests/composer.json (hunk 1 above)
Comment (yvesh) [inline tests/composer.json:23]: @pritalpatel But Puneet commited his composer.lock ;)

File: tests/acceptance/content.feature [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    Given Joomla CMS is installed
+    When Login into Joomla administrator with username "puneet" and password "1234"
+    Then I see administrator dashboard
+
+  Scenario: Create an Article
+    Given There is a Add Content link
+    When I fill mandatory fields for creating article
+      | field   | value                    |
+      | title   | My_Article               |
+      | content | This is my First Article |
```
Comment (javigomez) [inline tests/acceptance/content.feature:16]: very cool!

But I would only use tables when we want to repeat the same test with different values.

Fore example: create 10 articles:
- article1 | content1
- article2 | content2
- article3 | content3

see:
screen shot 2016-06-10 at 11 18 19

**Comment by javigomez** [comment]
I'm closing this pull since it was for serving as example for the others pulls already done.
